src/components/link/link.scss

```diff
@@ -1,16 +0,0 @@
-.carbon-link__icon {
-  margin-right: 5px;
-  position: relative;
-  top: -2px;
-
-  .carbon-icon__svg {
-    height: 16px;
-    vertical-align: middle;
-    width: 16px;
-  }
```

Comment:
I can't see these style in `style.js` files. Doesn't we still need them?

Reply:
Adding this style in the new styled-component approach means that the icons are not aligned with the links so I added other styes to achieve this